int16 xpos
uint8 angle